### PR TITLE
JBTM-2624 Update to support the admin commands in the karaf shell

### PR DESCRIPTION
**Pull request by zhfeng**
https://issues.jboss.org/browse/JBTM-2624

!XTS !BLACKTIE !QA_JTA !QA_JTS_JDKORB !QA_JTS_JACORB !PERF

### CODE REVIEW

File: scripts/hudson/narayana.sh [hunk 1]
```diff
@@ -290,7 +290,7 @@ function init_jboss_home {
 function osgi_tests {
   echo "#-1. OSGI Test"
   cd ${WORKSPACE}
-  ./build.sh -f osgi/jta/pom.xml -Parq-karaf-embedded clean test
+  #./build.sh -f osgi/jta/pom.xml -Parq-karaf-embedded clean integration-test
```
Comment (zhfeng) [inline scripts/hudson/narayana.sh]: just point that I have to disable the integration test temperately as the aquillian-container-karaf-embedded only support the karaf 2.3.3 and we are using the shell command with 4.x

**Comment by zhfeng** [comment]
I update to add a Heuristic example in the https://github.com/jbosstm/quickstart/pull/171
To run these commands, you need to replace the narayana version with 5.3.4.Final-SNAPSHOT in <karaf-4.1.0-SNAPSHOT>/system/org/apache/karaf/features/enterprise/4.1.0-SNAPSHOT/enterprise-4.1.0-SNAPSHOT-features.xml
When start the karaf, you need to run bundle:list to make sure 
```
114 | Active |  80 | 5.3.4.Final-SNAPSHOT | Narayana: OSGi Transaction Service Bundles - JTA
121 | Active |  80 | 5.3.4.Final-SNAPSHOT | Narayana Quickstarts: JBTM and OSGI
```
Run the Heuristic example
```
karaf@root()> testHeuristic 
testHeuristic
DummyXAResource XA_PREPARE [< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a8006a:a0c9:575e6590:74, node_name=1, branch_uid=0:ffffc0a8006a:a0c9:575e6590:75, subordinatenodename=null, eis_name=0 >]
Error executing command: javax.transaction.HeuristicMixedException
```
Show the heuristic record in the object stores
```
karaf@root()> narayana:refresh
karaf@root()> narayana:select StateManager/BasicAction/TwoPhaseCoordinator/AtomicAction
karaf@root()> narayana:ls
Transaction: jboss.jta:type=ObjectStore,itype=StateManager/BasicAction/TwoPhaseCoordinator/AtomicAction,uid=0_ffffc0a8006a_a0c9_575e6590_74
-----------------------------------
	AgeInSeconds=15
	Participant=false
	CreationTime=Tue, 14 Jun 2016 01:19:24 +0800
	Id=0:ffffc0a8006a:a0c9:575e6590:74
	Type=StateManager/BasicAction/TwoPhaseCoordinator/AtomicAction
	Participants:
		Participant: com.arjuna.ats.arjuna.tools.osb.mbean.LogRecordWrapper[jboss.jta:type=ObjectStore,itype=StateManager/BasicAction/TwoPhaseCoordinator/AtomicAction,uid=0_ffffc0a8006a_a0c9_575e6590_74,puid=0_ffffc0a8006a_a0c9_575e6590_76]
			HeuristicStatus=UNKNOWN
			Participant=true
			Status=HEURISTIC
			Type=/StateManager/AbstractRecord/XAResourceRecord
			Id=0:ffffc0a8006a:a0c9:575e6590:76

```
resolve the heuristic list and run recovery
```
karaf@root()> narayana:attach 0_ffffc0a8006a_a0c9_575e6590_74
karaf@root()> narayana:forget 0
karaf@root()> narayana-quickstart:testRecovery
[Periodic Recovery] DummyXAResourceRecovery Added DummyXAResource: 7fc35d2c-31b3-4cc1-a043-6e9790c80189_
[Periodic Recovery] DummyXAResourceRecovery returning list of DummyXAResources of length: 1
DummyXAResource XA_COMMIT  [< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a8006a:a0c9:575e6590:74, node_name=1, branch_uid=0:ffffc0a8006a:a0c9:575e6590:75, subordinatenodename=null, eis_name=0 >] with fault NONE
``` 
Refresh and the heuristic record is cleared
```
karaf@root()> narayana:refresh
karaf@root()> narayana:ls
```

File: scripts/hudson/narayana.sh (hunk 1 above)
Comment (mmusgrov) [inline scripts/hudson/narayana.sh]: OK. Di you raise a JIRA for that?

Comment (zhfeng) [inline scripts/hudson/narayana.sh]: https://issues.jboss.org/browse/JBTM-2689

Comment (mmusgrov) [inline scripts/hudson/narayana.sh]: Thanks

**Comment by mmusgrov** [comment]
@zhfeng Amos I missed your comment above that describes how to run the Heuristic example in the jbosstm/quickstart#171 - you need to update the quickstart README and add the information you have placed in this PR comment

**Comment by mmusgrov** [comment]
@zhfeng Did you see this comment and my comment on https://github.com/jbosstm/quickstart/pull/171

**Pull request merged**